kornia/augmentation/augmentation.py

```diff
@@ -259,111 +211,104 @@ class RandomGrayscale(AugmentationBase):
                   [-0.1717, -0.9023,  0.0819]]]])
     """
 
-    def __init__(self, p: float = 0.1, return_transform: bool = False, same_on_batch: bool = False) -> None:
-        super(RandomGrayscale, self).__init__(return_transform)
-        self.p = p
-        self.same_on_batch = same_on_batch
+    def __init__(self, return_transform: bool = False, same_on_batch: bool = False, p: float = 0.1) -> None:
+        super(RandomGrayscale, self).__init__(p=p, return_transform=return_transform, same_on_batch=same_on_batch)
 
     def __repr__(self) -> str:
-        repr = f"(p={self.p}, return_transform={self.return_transform}, same_on_batch={self.same_on_batch})"
-        return self.__class__.__name__ + repr
+        return self.__class__.__name__ + f"({super().__repr__()})"
 
     def generate_parameters(self, batch_shape: torch.Size) -> Dict[str, torch.Tensor]:
-        return rg.random_prob_generator(batch_shape[0], self.p, self.same_on_batch)
+        return dict()
```

Comment:
is there any way that in case we don't override the parent class returns an empty dict ?

Reply:
I think having this can be safer with knowing it is implemented but expected to be empty.